util-images/network/netperfbenchmark/pkg/worker/util.go

```diff
@@ -0,0 +1,174 @@
+func ParseResult(protocol string, result []string, testDuration string) ([]float64, error) {
+	klog.Infof("Parsing response for %v", protocol)
+	switch protocol {
+	case ProtocolTCP:
+		return parseIperfResponse(result, testDuration, ProtocolTCP, tcpMetricsCount, zeroFormatTCP, fractionZeroFormatTCP, iperfTCPFunction), nil
+	case ProtocolUDP:
+		return parseIperfResponse(result, testDuration, ProtocolUDP, udpMetricsCount, zeroFormatUDP, fractionZeroFormatUDP, iperfUDPFunction), nil
+	case ProtocolHTTP:
+		return parseSiegeResponse(result), nil
+	default:
+		return nil, errors.New("invalid protocol: " + protocol)
+	}
+}
+
+func parseIperfResponse(result []string, testDuration string, protocol string, metricCount int, zeroFormat string, fractionZeroFormat string, operators []string) []float64 {
+	klog.Info("Parsing iperf response")
+	aggregatedResult := make([]float64, 0, metricCount)
+	unitRegex := regexp.MustCompile(`%|\[\s+|\]\s+|KBytes\s+|KBytes/sec\s*|sec\s+|pps\s*|ms\s+|/|\(|\)\s+`)
+	multiSpaceRegex := regexp.MustCompile(`\s+`)
+	hyphenSpaceRegex := regexp.MustCompile(`\-\s+`)
+	count := 0
+	sessionID := make(map[string]bool)
+	for _, op := range result {
+		formattedString := hyphenSpaceRegex.ReplaceAllString(multiSpaceRegex.ReplaceAllString(unitRegex.ReplaceAllString(op, " "), " "), "-")
+		if !strings.Contains(formattedString, zeroFormat+"-"+testDuration+fractionZeroFormat) {
+			continue
+		}
+		split := strings.Split(formattedString, " ")
+		//for bug in iperf tcp
+		//if the record is for the complete duration of run
+		if len(split) >= metricCount+1 && "SUM" != split[1] && split[2] == zeroFormat+"-"+testDuration+fractionZeroFormat {
+			if _, ok := sessionID[split[1]]; ok {
+				continue
+			}
+			for i, v := range split {
+				if i == 1 {
+					sessionID[v] = true
+					continue
+				}
+				//first index and hte last is ""
+				if i == 0 || i == 2 || i == metricCount+3 {
+					continue
+				}
+				tmp, err := strconv.ParseFloat(v, 64)
+				if err != nil {
+					klog.Errorf("Conversion error %v", err)
+				}
+				if len(aggregatedResult) < metricCount {
+					aggregatedResult = append(aggregatedResult, tmp)
+				} else {
+					switch operators[i] {
+					case "Sum":
+						aggregatedResult[i-3] = tmp + aggregatedResult[i-3]
+					case "Avg":
+						aggregatedResult[i-3] = (float64(count)*tmp + aggregatedResult[i-3]) / (float64(1 + count))
+					case "Min":
+						aggregatedResult[i-3] = math.Min(tmp, aggregatedResult[i-3])
+					case "Max":
+						aggregatedResult[i-3] = math.Max(tmp, aggregatedResult[i-3])
+					}
+				}
+				count++
+			}
+		}
+	}
+	klog.Infof("Final output: %v", aggregatedResult)
+	return aggregatedResult
+}
+
+func parseSiegeResponse(result []string) []float64 {
+	canAppend := false
+	aggregatedResult := make([]float64, 0, httpMetricsCount)
+	multiSpaceRegex := regexp.MustCompile(`\s+`)
+	for _, op := range result {
+		if canAppend != true && strings.HasPrefix(op, "Transactions:") {
```

Comment:
IIUC, you're trying to remove all lines before "Transactions:" and all after "Shortest transaction".

It would be much cleaner, to first trim the "result" value and do the actually parsing here.
Something like:

result = trimSiegeResponse(result)

trimSiegeResponse {
  var beginIndex, endIndex int
  for beginIndex := 0; beginIndex < len(result) && !strings.HasPrefix("Transactions":); beginIndex++ {}
  if beginIndex == len(result) {
    return nil
  }
  beingIndex++
  for endIndex := beginIndex; endIndex < len(result) && !strings.hasPrefix("Short..."); endIndex++ {}
  return result[beginIndex:endIndex]
}

Reply:
Modified